server/storage/backend/read_tx.go

```diff
@@ -83,13 +93,62 @@ func (baseReadTx *baseReadTx) UnsafeRange(bucketType Bucket, key, endKey []byte,
 		limit = math.MaxInt64
 	}
 	if limit > 1 && !bucketType.IsSafeRangeBucket() {
-		panic("do not use unsafeRange on non-keys bucket")
+		panic("do not use UnsafeRange on non-keys bucket")
+	}
+
+	// ConcurrentReadTxNoCopyMode only for Key bucket
+	if baseReadTx.bufferNoCopy && endKey == nil {
+		// Single key search. Key exists either bolt db or read buffer
+		k1, v1 := baseReadTx.unsafeRangeDB(bucketType, key, endKey, limit)
+		if len(k1) != 0 {
+			return k1, v1
+		}
+		k2, v2 := baseReadTx.unsafeRangeBuff(bucketType, key, endKey, limit-int64(len(k1)))
+		if len(k2) != 0 {
+			// If rate of buffRangeHitCounter/rangeTotalCounter is high, we should not use ConcurrentReadTxNoCopyMode
+			buffRangeHitCounter.Inc()
+		}
```

Comment:
I do not see any need to handle single key range separately, and we should also try to get the k/v from the buf firstly.